pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,537 @@
+            alpha = self.model.register_rv(alpha_prior, name="alpha")
+
+            # churn priors
+            s = self.model.register_rv(s_prior, name="s")
+            beta = self.model.register_rv(beta_prior, name="beta")
+
+            ParetoNBD(
+                name="likelihood",
+                r=r,
+                alpha=alpha,
+                s=s,
+                beta=beta,
+                T=self._T,
+                observed=np.stack((self._recency, self._frequency), axis=1),
+                dims="customer_id",
+            )
+
+    def _process_priors(self, r_prior, alpha_prior, s_prior, beta_prior):
+        # priors for purchase rate
+        if r_prior is None:
+            r_prior = pm.Weibull.dist(alpha=10, beta=1)
+        else:
+            self._check_prior_ndim(r_prior)
+        if alpha_prior is None:
+            alpha_prior = pm.Weibull.dist(alpha=10, beta=10)
+        else:
+            self._check_prior_ndim(alpha_prior)
+
+        # hyper priors for churn rate
+        if s_prior is None:
+            s_prior = pm.Weibull.dist(alpha=10, beta=1)
+        else:
+            self._check_prior_ndim(s_prior)
+        if beta_prior is None:
+            beta_prior = pm.Weibull.dist(alpha=10, beta=10)
+        else:
+            self._check_prior_ndim(beta_prior)
+
+        return super()._process_priors(r_prior, alpha_prior, s_prior, beta_prior)
+
+    def _unload_params(self) -> Tuple[xarray.DataArray]:
+        """Utility function retrieving posterior parameters for predictive methods"""
+        return tuple([self.fit_result.posterior[param] for param in self._params])
+
+    # TODO: Convert to list comprehension to support covariates?
+    def _process_customers(
+        self,
+        customer_id: Union[np.ndarray, pd.Series],
+        frequency: Union[np.ndarray, pd.Series],
+        recency: Union[np.ndarray, pd.Series],
+        T: Union[np.ndarray, pd.Series],
+    ) -> Tuple[xarray.DataArray]:
+        """Utility function assigning default customer arguments
+        for predictive methods and converting to xarrays.
+        """
+        if customer_id is None:
+            customer_id = self._customer_id
+        if frequency is None:
+            frequency = self._frequency
+        if recency is None:
+            recency = self._recency
+        if T is None:
+            T = self._T
+
+        return to_xarray(customer_id, frequency, recency, T)
+
+    @staticmethod
+    def _logp(
+        r: np.ndarray,
+        alpha: np.ndarray,
+        s: np.ndarray,
+        beta: np.ndarray,
+        x: xarray.DataArray,
+        t_x: xarray.DataArray,
+        T: xarray.DataArray,
+    ) -> xarray.DataArray:
+        """
+        Utility function for using ParetoNBD log-likelihood in predictive methods.
+        """
+        # Add one dummy dimension to the right of the scalar parameters, so they broadcast with the `T` vector
+        pareto_dist = ParetoNBD.dist(
+            r=r.values[..., None],
```

Comment:
Are you sure r (and other variables are `numpy` arrays?). Mypy is complaining about 
```python
"ndarray[Any, Any]" has no attribute "values"
```
Should they be pandas `Series`?